libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -146,3 +153,23 @@ func (p *addHostMetadata) String() string {
 	return fmt.Sprintf("%v=[netinfo.enabled=[%v], cache.ttl=[%v]]",
 		processorName, p.config.NetInfoEnabled, p.config.CacheTTL)
 }
+
+func skipAddingHostMetadata(event *beat.Event) bool {
+	// If host fields exist(besides host.name added by libbeat) in event, skip add_host_metadata.
+	hostFields, err := event.Fields.GetValue("host")
+	hostFieldsMap := hostFields.(common.MapStr)
+	if err == nil && hostFields != nil && len(hostFieldsMap) >= 1 {
+		if hasName, _ := hostFieldsMap.HasKey("name"); hasName {
+			// other host fields exist on top of host.name, skip add_host_metadata.
+			if len(hostFieldsMap) > 1 {
+				return true
+			}
+		} else {
+			// host.name does not exist but other host fields exist, skip add_host_metadata.
+			if len(hostFieldsMap) > 0 {
+				return true
+			}
+		}
+	}
+	return false
```

Comment:
Nit. I think this can be simplified if we return earlier. 
```suggestion
	// Don't skip if there are no fields...
	if err != nil || hostFields == nil || len(hostFieldsMap) == 0 {
		return false
	}
	// or if "name" is the only field
	hasName, _ := hostFieldsMap.HasKey("name");
	if hasName && len(hostFieldsMap) == 1
		return false
	}
	return true
```

Reply:
Thank you!! Yes this way is a lot simpler!!